src/Routers/CloudCodeRouter.js

```diff
@@ -53,6 +54,7 @@ export class CloudCodeRouter extends PromiseRouter {
       middleware.promiseEnforceMasterKeyAccess,
       CloudCodeRouter.deleteJob
     );
+    this.route('GET', '/securityChecks', middleware.promiseEnforceMasterKeyAccess, securityChecks);
   }
```

Comment:
What do you think about removing the endpoint also if the options is not enabled ?

Reply:
What would be the argument for removing the endpoint?

Reply:
i think we should avoid exposing endpoints (even if it's protected by master key), if the feature is disabled ! Here, it is even more important because it provides advanced security feedback from the server.

Reply:
Not related but i don't know if notifications works correctly: Also @mtrezza could you check your parse community inbox ? I would like to open a security advisory. But it seems that i do not have rights on the repo explain the issue.

Reply:
> i think we should avoid exposing endpoints (even if it's protected by master key), if the feature is disabled ! Here, it is even more important because it provides advanced security feedback from the server.

I expect that we will enable this feature by default in the future, and that most deployments will have this enabled. A production system should really prevent unauthorized access using firewall rules, we can add that to the docs as recommendation. The downside I see is that removing an endpoint means returning 404 but the correct response is really 403, either by Parse Server, or by a firewall.

Reply:
You are right the endpoint should throw 403 if option not enabled !